src/structures/Guild.js

```diff
@@ -347,17 +346,15 @@ class Guild {
   /**
    * Fetches all the members in the guild, even if they are offline. If the guild has less than 250 members,
    * this should not be necessary.
-   * @param {string} [query=''] An optional query to provide when fetching members
-   * @returns {Promise<Guild>}
+   * @param {string} [query=''] Limit fetch to members with similar usernames
+   * @returns {Promise<Collection<GuildMember>>}
    */
   fetchMembers(query = '') {
     return new Promise((resolve, reject) => {
-      if (this._fetchWaiter) throw new Error(`Already fetching guild members in ${this.id}.`);
       if (this.memberCount === this.members.size) {
-        resolve(this);
+        resolve(this.members);
```

Comment:
There's no way anyone actually uses `fetchMembers` and relies on the result being a guild.